wild_lib/src/elf_writer.rs

```diff
@@ -1661,6 +1753,82 @@ fn apply_relocation(
     Ok(next_modifier)
 }
 
+fn apply_debug_relocation(
+    object_layout: &ObjectLayout,
+    offset_in_section: u64,
+    rel: &elf::Rela,
+    section_info: SectionInfo,
+    layout: &Layout,
+    section_tombstone_values: &[u64],
+    out: &mut [u8],
+) -> Result<()> {
+    let section_address = section_info.section_address;
+    let place = section_address + offset_in_section;
+    let _span = tracing::trace_span!("debug relocation", address = place).entered();
+
+    let e = LittleEndian;
+    let symbol_index = rel
+        .symbol(e, false)
+        .context("Unsupported absolute relocation")?;
+    let sym = object_layout.object.symbol(symbol_index)?;
+    let section_index = object_layout.object.symbol_section(sym, symbol_index)?;
+
+    let addend = rel.r_addend.get(e) as u64;
+    let r_type = rel.r_type(e, false);
+    let rel_info = RelocationKindInfo::from_raw(r_type)?;
+
+    let resolution = layout
+        .merged_symbol_resolution(object_layout.symbol_id_range.input_to_id(symbol_index))
+        .or_else(|| {
+            section_index
+                .and_then(|section_index| object_layout.section_resolutions[section_index.0])
+        });
+
+    let value = if let Some(resolution) = resolution {
+        match rel_info.kind {
+            RelocationKind::Absolute => resolution.value_with_addend(
+                addend,
+                symbol_index,
+                object_layout,
+                &layout.merged_strings,
+                &layout.merged_string_start_addresses,
+            )?,
+            RelocationKind::DtpOff => resolution
+                .value()
+                .wrapping_sub(layout.tls_end_address())
+                .wrapping_add(addend),
+            kind => bail!("Unsupported debug relocation kind {kind:?}"),
+        }
+    } else if let Some(section_index) = section_index {
+        match object_layout.sections[section_index.0] {
+            SectionSlot::MergeStrings(..) => get_merged_string_output_address(
+                symbol_index,
+                addend,
+                object_layout.object,
+                &object_layout.sections,
+                &layout.merged_strings,
+                &layout.merged_string_start_addresses,
+                false,
+            )?
+            .context("Cannot get merged string offset for a debug info section")?,
+            SectionSlot::Discard | SectionSlot::Unloaded(..) => {
+                section_tombstone_values[section_index.0]
```

Comment:
Sorry I didn't notice this pre-submission, but is this right? It looks like you're checking if a section that was discarded or not loaded is one of the two special debug sections. But won't those debug sections be unconditionally loaded? I'd have thought that the tombstone value would value would depend on the section to which the relocation applied (source) rather than the section to which the relocation referred (target).

If that's the case, then you wouldn't need to create a `Vec` of tombstone values because all the tombstones for the section we're currently processing would have the same tombstone.

Reply:
Yep, good catch. The tombstone must be related to the place where we apply relocations (source section).
Please merge: #115.